GameServ/GameHost.cpp

```diff
@@ -37,6 +37,21 @@ agemap_t s_ages;
 #define SEND_REPLY(msg, result) \
     msg->m_client->m_channel.putMessage(result)
 
+#define DM_SENDBUF(client) \
+    _msgbuf->ref(); \
+    client->m_broadcast.putMessage(e_GameToCli_PropagateBuffer, _msgbuf)
+
+#define DM_UNREFBUF() \
+    _msgbuf->unref()
+
+#define DM_WRITEBUF(msg) \
+    DS::BufferStream* _msgbuf = new DS::BufferStream(); \
+    _msgbuf->write<uint32_t>(msg->type()); \
+    _msgbuf->write<uint32_t>(0); \
+    MOUL::Factory::WriteCreatable(_msgbuf, msg); \
+    _msgbuf->seek(4, SEEK_SET); \
+    _msgbuf->write<uint32_t>(_msgbuf->size() - 8)
+
 #define DM_WRITEMSG(host, msg) \
```

Comment:
Good question. It apparently is -- that means I didn't move all client communication to the client thread as advertised (oops). That needs to be addressed. I'll get on it as soon as I fix my VM. :stuck_out_tongue: